### ISSUE TITLE
[Feature Request] Have a signal to force the train to leave a provider/requester stop

### ISSUE THREAD

**Issue by KaidenP** [Yousei9/Logistic-Train-Network#310]
**TL;DR & What?**
Have a signal fed into the LTN input or train stop to force the train to leave a provider/requester stop.

**Why ?**
I believe this would be a nice feature for edge cases, for example if you have multiple provider stops fed by the same inventory, and that inventory runs dry due to two trains coming at the same time and there wasn't enough available buffer, or other reasons you may want to evacuate the station early.

I can see very limited use cases for the requester side, maybe if you had a mod or some other way of detecting if a wall was breached, and you want the train to evacuate to prevent it from being destroyed?

Personally, I have 2 provider stops sharing the same inventory of many (>2) different mall type items. I have two stops to increase short term throughput if multiple requests for different items happen at the same time. I have not experienced an issue as of yet, but I can foresee a possible issue where repetitive requests of the same item almost completely depletes the buffer, leaves enough that it can still satisfy one train, but two requests happen at the same time for the same item across the two stations. leaving the trains to wait until the timeout occurs.

A solution would be to lower the timeout, but I have other stops that have a necessarily long loading time that I would prefer to finish.

An alternate feature request would be to have a signal that specifies how long the train will wait before timing out and leaving (compared to the global map setting timeout), but I suspect that would be harder to implement, and would offer less flexibility.

**How this can be implemented**
![image](https://user-images.githubusercontent.com/11036039/196569153-3605e49a-6a01-44e0-ab31-dafa4039063d.png)
Simply automatically add a signal to the train schedule so that when the train detects it, it just leaves

**Comment by Yousei9** [comment]
Enable Circuit Conditions in LTN settings.
See Manual: https://forums.factorio.com/viewtopic.php?f=214&t=51072
